secantTests/LoadingScreenTests.swift

```diff
@@ -0,0 +1,239 @@
+        mockServices.keyStorage = stub
+        let loadingViewModel = LoadingScreenViewModel(services: mockServices)
+                
+        let expected = LoadingScreenViewModel.LoadingResult.credentialsFound
+        let result = loadingViewModel.load()
+        
+        XCTAssertTrue(stub.called)
+        switch result {
+        case .failure(let error):
+            XCTFail("found error \(error.localizedDescription)")
+        case .success(let res):
+            XCTAssertEqual(expected, res)
+        }
+    }
+    
+    func testLoadReturnsErrorWhenLoadingFails() throws {
+        let mockServices = MockServices()
+        let stub = KeysPresentStub(returnBlock: {
+            throw KeyStoringError.uninitializedWallet
+        })
+        mockServices.keyStorage = stub
+        let loadingViewModel = LoadingScreenViewModel(services: mockServices)
+
+        let result = loadingViewModel.load()
+        XCTAssertTrue(stub.called)
+        switch result {
+        case .failure:
+            XCTAssertTrue(true)
+        case .success(let res):
+            XCTFail("case succeeded when testing failure - result: \(res)")
+        }
+    }
+    
+    // MARK: LoadingScreen View Tests
+    
+    func testProceedToHomeIsCalledWhenCredentialsAreFound() throws {
+        let mockServices = MockServices()
+        let stub = KeysPresentStub(returnBlock: {
+            true
+        })
+        mockServices.keyStorage = stub
+        let loadingViewModel = LoadingScreenViewModel(services: mockServices)
+
+        let spyRouter = LoadingScreenRouterSpy(fulfillment: {
+        })
+        
+        loadingViewModel.callRouter(spyRouter, with: loadingViewModel.load())
+        XCTAssertTrue(spyRouter.proceedToHomeCalled)
+    }
+    
+    func testProceedToWelcomeIsCalledWhenCredentialsAreNotFound() throws {
+        let mockServices = MockServices()
+        let stub = KeysPresentStub(returnBlock: {
+            false
+        })
+        mockServices.keyStorage = stub
+        let loadingViewModel = LoadingScreenViewModel(services: mockServices)
+
+        let spyRouter = LoadingScreenRouterSpy(fulfillment: {
+        })
+        
+        loadingViewModel.callRouter(spyRouter, with: loadingViewModel.load())
+        XCTAssertTrue(spyRouter.proceedToWelcomeCalled)
+    }
+    
+    func testFailWithErrorIsCalledWhenKeyStoringFails() throws {
+        let mockServices = MockServices()
+        let stub = KeysPresentStub(returnBlock: {
+            throw KeyStoringError.alreadyImported
+        })
+        mockServices.keyStorage = stub
+        let loadingViewModel = LoadingScreenViewModel(services: mockServices)
+
+        let spyRouter = LoadingScreenRouterSpy(fulfillment: {
+        })
+        
+        loadingViewModel.callRouter(spyRouter, with: loadingViewModel.load())
+        XCTAssertTrue(spyRouter.failWithErrorCalled)
+    }
+}
+
+class LoadingScreenRouterSpy: LoadingScreenRouter {
+    init(fulfillment: @escaping () -> Void) {
```

Comment:
init functions should come after the property declarations for the structure we've defined for classes.

Reply:
I'm going to make a post it with our own rules LOL